frontend/src/metabase/nav/containers/MainNavbar/SidebarItems/DwhUploadCSV/DwhUploadCSV.tsx

```diff
@@ -0,0 +1,94 @@
+import { type ChangeEvent, useCallback, useRef, useState } from "react";
+import { t } from "ttag";
+
+import {
+  type CollectionOrTableIdProps,
+  ModelUploadModal,
+} from "metabase/collections/components/ModelUploadModal";
+import type { OnFileUpload } from "metabase/collections/types";
+import { UploadInput } from "metabase/components/upload";
+import { useToggle } from "metabase/hooks/use-toggle";
+import { useDispatch } from "metabase/lib/redux";
+import {
+  type UploadFileProps,
+  uploadFile as uploadFileAction,
+} from "metabase/redux/uploads";
+import { Box, Button, Icon } from "metabase/ui";
+
+import { trackDWHUploadCSVClicked } from "./analytics";
+
+export const DwhUploadCSV = () => {
+  const [
+    isModelUploadModalOpen,
+    { turnOn: openModelUploadModal, turnOff: closeModelUploadModal },
+  ] = useToggle(false);
+
+  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+
+  const dispatch = useDispatch();
+  const uploadFile = useCallback(
+    ({ file, modelId, collectionId, tableId, uploadMode }: UploadFileProps) =>
+      dispatch(
+        uploadFileAction({ file, modelId, collectionId, tableId, uploadMode }),
+      ),
+    [dispatch],
+  );
+
+  const handleFileUpload = useCallback<OnFileUpload>(
+    (uploadFileArgs: CollectionOrTableIdProps) => {
+      const { collectionId, tableId } = uploadFileArgs;
+      if (uploadedFile && (collectionId || tableId)) {
+        closeModelUploadModal();
+        uploadFile({
+          file: uploadedFile,
+          ...uploadFileArgs,
+        });
+      }
+    },
+    [uploadFile, uploadedFile, closeModelUploadModal],
+  );
+
+  const handleFileInput = (event: ChangeEvent<HTMLInputElement>) => {
+    trackDWHUploadCSVClicked();
+    const file = event.target.files?.[0];
+
+    if (file !== undefined) {
```

Comment:
nonblocking suggestion / thinking out loud:

An alternative, perhaps cleaner, but not necessarily better:

```suggestion
    if (file) {
```